### PR TITLE
Marketplace Reviews: Add the Reviews Modal to Plugin Details page

### PR DESCRIPTION
**Pull request by WBerredo**
Fixes https://github.com/Automattic/dotcom-forge/issues/4843

## Proposed Changes

* Rename the old component called ReviewsModal to ReviewModal (this component refers to a single review added or updated)
* Create a new component a ReviewsModal to display the reviews of a given product
* Link it to the `Read all reviews` button on the Plugin Details page
* Create different states for this component as below:

| With reviews  | No reviews (with subscription) | No reviews (without subscription)|
| ------------- | ------------- |------------- |
|CleanShot 2023-12-19 at 12 45 13@2x|CleanShot 2023-12-19 at 12 30 18@2x|CleanShot 2023-12-19 at 12 29 21@2x|

## Out of scope
* Display the user avatar - #85503
* Always display the user review first - https://github.com/Automattic/wp-calypso/pull/85549
* Warn the user if the review is pending approval - https://github.com/Automattic/dotcom-forge/issues/5029
* Update the button to also support theme purchase - https://github.com/Automattic/dotcom-forge/issues/5024
* Add new review section - https://github.com/Automattic/dotcom-forge/issues/5023
* Update review - https://github.com/Automattic/dotcom-forge/issues/5026

## Testing Instructions

* On this branch
* Go to a plugin details page with reviews. Ex: `/plugins/woocommerce-bookings/`
* On the bottom, click on Read all reviews
* Check if all reviews are displayed 
* Check if plugin without reviews are shown as above
* Check if the button to purchase is displayed if you don't have an active subscription for a given plugin
* Check if the `Leave review` button is displayed only on plugins you have an active subscription

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/marketplace/components/reviews-modal/styles.scss [hunk 1]
```diff
@@ -1,45 +1,78 @@
-@import "@automattic/typography/styles/variables.scss";
+$border-color: #eee;
 
-.marketplace-reviews-modal__card {
-	padding: 0 1rem;
-	box-shadow: none;
-}
+.marketplace-reviews-modal {
+	max-width: 850px;
```
Comment (gcsecsey) [inline client/my-sites/marketplace/components/reviews-modal/styles.scss]: If there's no long reviews for a plugin yet (eg. with `/plugins/woocommerce-subscriptions`), the modal looks a bit cramped, especially with the wide CTA:
CleanShot 2023-12-20 at 13 11 00@2x

Maybe we should add a `min-width` here too to compensate this.

**Review by gcsecsey — COMMENTED** [review]
LGTM and works as described, I only have a minor comment regarding the layout of the modal

File: client/my-sites/marketplace/components/reviews-cards/index.tsx [hunk 2]
```diff
@@ -46,7 +48,7 @@ export const MarketplaceReviewsCards = ( props: ProductProps ) => {
 						className="is-link"
 						borderless
 						primary
-						onClick={ () => alert( 'Not implemented yet!' ) }
+						onClick={ () => props.showMarketplaceReviews && props.showMarketplaceReviews() }
```
Comment (gcsecsey) [inline client/my-sites/marketplace/components/reviews-cards/index.tsx:51]: Can we update the button to use the same callback in the PluginDetailsHeader here:
https://github.com/Automattic/wp-calypso/blob/0c81f27d2f0858d8973d7f0564a8327235bfe1ed/client/my-sites/plugins/plugin-details-header/index.jsx#L93-L104

Or should this be handled in a separate PR?

Comment (WBerredo) [reply client/my-sites/marketplace/components/reviews-cards/index.tsx:51]: I would say this should come on the PRs that change the Plugin Details page layout.

File: client/my-sites/marketplace/components/reviews-modal/styles.scss (hunk 1 above)
Comment (WBerredo) [reply client/my-sites/marketplace/components/reviews-modal/styles.scss]: I've added a few intermediary steps here, let me know how that works for you: https://github.com/Automattic/wp-calypso/pull/85496/commits/4b7e00298c74c2585554457bba79e74a8bf9580b

Comment (WBerredo) [reply client/my-sites/marketplace/components/reviews-modal/styles.scss]: Also added a gap to avoid them to get too close https://github.com/Automattic/wp-calypso/pull/85496/commits/135c84af3865e0bb3a5964438dd86ae2b30c81f7

Comment (gcsecsey) [reply client/my-sites/marketplace/components/reviews-modal/styles.scss]: Brilliant, thanks so much for iterating on this!

Comment (gcsecsey) [reply client/my-sites/marketplace/components/reviews-modal/styles.scss]: It looks great now on all breakpoints 👍 
CleanShot 2023-12-20 at 15 57 42@2x

`<960`:
CleanShot 2023-12-20 at 15 58 17@2x

`<660`:
CleanShot 2023-12-20 at 15 58 23@2x

**Review by gcsecsey — APPROVED** [review]
Thanks for the updates, let's merge this once CI is happy too.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10445398

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @WBerredo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10445398

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @WBerredo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10445398).
